### ISSUE TITLE
Implement other variations of FWA

### ISSUE THREAD

**Issue by tsimafei-markhel** [tsimafei-markhel/Fireworks.NET#6]
Blocked by #4.
I can create separate issue for each variation if needed.

--
PDFs can be found by the [link](http://www.cil.pku.edu.cn/publications/).
- 2012:
 - Y. Pei, S.Q. Zheng, Y. Tan and Hideyuki Takagi, "An Empirical Study on Influence of Approximation Approaches on Enhancing Fireworks Algorithm", IEEE International Conference on System, Man and Cybernetics (SMC 2012), Seoul, Korea. October 14-17, 2012.
- 2013:
 - Y. Tan, C. Yu, S.Q. Zheng and K. Ding "Introduction to Fireworks Algorithms", International Journal of Swarm Intelligence Research (IJSIR), October-December 2013, vol. 4, No. 4, pp. 39-71.
 - K. Ding, S.Q. Zheng and Y. Tan, "A GPU-based Parallel Fireworks Algorithm for Optimization", ACM Genetic and Evolutionary Computation Conference (GECCO 2013) - a recombination of 22nd International Conference on Genetic Algorithms (ICGA) and the 18th Annual Genetic Programming Conference (GP), Amsterdam, The Netherlands, July 06-10, 2013. pp. 1-8.
 - Shaoqiu Zheng, Andreas Janecek and Ying Tan, "Enhanced Fireworks Algorithm", 2013 IEEE Congress on Evolutionary Computation, (IEEE CEC 2013), June 20-23, Fiesta Americana Grand Coral Beach Hotel, Cancun, Mexico, pp. 10-19.
- 2014:
 - S.Q. Zheng, Andreas Janecek, J.Z. Li, and Y. Tan, "Dynamic Search in Fireworks Algorithm", 2014 IEEE World Conference on Computational Intelligence (IEEE WCCI'2014) - IEEE Congress on Evolutionary Computation (CEC'2014), July 07-11, 2014, Beijing International Convention Center (BICC), Beijing, China, pp. 3222-3229.
 - J.Z. Li, S.Q. Zheng, and Y. Tan, "Adaptive Fireworks Algorithm", 2014 IEEE World Conference on Computational Intelligence (IEEE WCCI'2014) - IEEE Congress on Evolutionary Computation (CEC'2014), July 07-11, 2014, Beijing International Convention Center (BICC), Beijing, China, pp. 3214-3221.
 - C. Yu, and Y. Tan, "Improving Enhanced Fireworks Algorithm with Differential Mutation", The 2014 IEEE International Conference on Systems, Man, and Cybernetics, October 5-8, 2014,Paradise Point Resort and Spa, San Diego, California, USA. pp. 270-275.

--
Cannot find PDFs for the following articles (maybe they should be implemented, too):
- J.H. Liu, S.Q. Zheng and Y. Tan, "The Improvement on Controlling Exploration and Exploitation of Firework Algorithm", The Fourth International Conference on Swarm Intelligence (ICSI 2013), Harbin, China, June 12-15, 2013. Springer, LNCS 7928, pp. 11-23.
- S.Q. Zheng, L. Liu, C. Yu, J.Z. Li, and Y. Tan, "Fireworks Algorithm and Its Variants for Solving ICSI 2014 Competition Problems", The Fifth International Conference on Swarm Intelligence (ICSI 2014), Hefei, China, October 17-20, 2014. Springer, LNCS 8794, pp. 442-451.
